DownFall/Views/FillableBar.swift

```diff
@@ -47,26 +49,51 @@ class FillableBar: SKSpriteNode {
 
     
     private lazy var fill: SKShapeNode = {
-        // Subtract the line widths from the actual fill size height
-        let height = contentView.frame.height - 2*Style.FillableBar.lineWidth
         
-        // Determine how long the filled portion of the bar should be
-        let widthRatio: CGFloat = CGFloat(viewModel.progress) / CGFloat(viewModel.total)
-        let width = barOutline.frame.width * widthRatio - 3*Style.FillableBar.lineWidth
-        
-        let size = CGSize(width: width,
-                          height: height)
-        // create the shape
-        var fill = SKShapeNode(rect: CGRect(origin: contentView.frame.origin, size: size), cornerRadius: Style.FillableBar.cornerRadius)
-        
-        // add the color
-        fill.color = viewModel.fillColor
-        
-        // set the position
-        fill.position = CGPoint(x: Style.FillableBar.lineWidth,
-                                y: Style.FillableBar.lineWidth)
-        
-        return fill
+        if self.viewModel.horiztonal {
+            // Subtract the line widths from the actual fill size height
+            let height = contentView.frame.height - 2*Style.FillableBar.lineWidth
+            
+            // Determine how long the filled portion of the bar should be
+            let widthRatio: CGFloat = CGFloat(viewModel.progress) / CGFloat(viewModel.total)
+            let width = barOutline.frame.width * widthRatio - 3*Style.FillableBar.lineWidth
+            
+            let size = CGSize(width: width,
+                              height: height)
+            // create the shape
+            var fill = SKShapeNode(rect: CGRect(origin: contentView.frame.origin, size: size), cornerRadius: Style.FillableBar.cornerRadius)
+            
+            // add the color
+            fill.color = viewModel.fillColor
+            
+            // set the position
+            fill.position = CGPoint(x: Style.FillableBar.lineWidth,
+                                    y: Style.FillableBar.lineWidth)
+            
+            return fill
+        } else {
+            // Subtract the line widths from the actual fill size height
+            let width = contentView.frame.width - 2*Style.FillableBar.lineWidth
+            
+            // Determine how long the filled portion of the bar should be
+            let heightRatio: CGFloat = CGFloat(viewModel.progress) / CGFloat(viewModel.total)
+            let height = barOutline.frame.height * heightRatio - 3*Style.FillableBar.lineWidth
+            
```

Comment:
These are the only lines with logic specific to the direction horizontal or vertical.  The rest of the lines can be removed and placed after the if else